### PR TITLE
qemu: only set highmem=off for darwin if qemu version is below 7.0 or memory is below 3GB

### LINKED ISSUE #14273: ERROR: qemu-system-aarch64: Addressing limited to 32 bits
### What Happened?

When starting minikube [v1.26.0-beta.1](https://github.com/kubernetes/minikube/releases/tag/v1.26.0-beta.1) with the following configuration, it fails to start.

```yaml
- cpus: 8
- driver: qemu2
- memory: 32768
```

Log file is attached though the error indicates a 32-bit memory address limitation.  My host configuration is...
* OS: macOS Monterey (12.4)
* CPU: Apple M1 Ultra
* Memory: 128 GB

**NOTE:** I wasn't able to select `qemu2` as the driver below on this form.

### Attach the log file

[logs.txt](https://github.com/kubernetes/minikube/files/8835280/logs.txt)

### Operating System

macOS (Default)

### Driver

_No response_

### PR DESCRIPTION
**Pull request by sharifelgamal**
qemu 7.0.0 fixes an issue that required highmem=off to be specified when using hardware acceleration in aarch64, so we should only specify it for older versions of qemu or if there isn't enough memory allocation to make highmem work.

Fixes #14273 

Related #14146

### CODE REVIEW

File: pkg/minikube/registry/drvs/qemu2/qemu2.go [hunk 1]
```diff
@@ -122,9 +122,12 @@ func configure(cc config.ClusterConfig, n config.Node) (interface{}, error) {
 		qemuCPU = "cortex-a72"
 		// highmem=off needed, see https://patchwork.kernel.org/project/qemu-devel/patch/20201126215017.41156-9-agraf@csgraf.de/#23800615 for details
 		if runtime.GOOS == "darwin" {
-			qemuMachine = "virt,highmem=off"
```
Comment (mprimeaux) [inline pkg/minikube/registry/drvs/qemu2/qemu2.go:125]: Should we use "+=" here also?

**Review by mprimeaux — COMMENTED** [review]
Just a minor change but nothing to hold back the PR.

File: pkg/minikube/registry/drvs/qemu2/qemu2.go (hunk 1 above)
Comment (sharifelgamal) [reply pkg/minikube/registry/drvs/qemu2/qemu2.go:126]: yeah, new commit fixes this

**Review by mprimeaux — APPROVED** [review]
LGTM

**Review by spowelljr — APPROVED** [review]

**Pull request merged**